hypertrace-gradle-java-convention-plugin/src/main/java/org/hypertrace/gradle/java/convention/JavaConventionExtension.java

```diff
@@ -0,0 +1,17 @@
+package org.hypertrace.gradle.java.convention;
+
+import javax.inject.Inject;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.Property;
+import org.gradle.jvm.toolchain.JavaLanguageVersion;
+
+public class JavaConventionExtension {
+  static final String EXTENSION_NAME = "javaConvention";
+  Property<JavaLanguageVersion> languageVersion;
```

Comment:
```suggestion
  public final Property<JavaLanguageVersion> languageVersion;
```

(public assuming you want it to be settable from a project)